### PR TITLE
Feature/slop 56/edit blog post

### PR DESCRIPTION
**Pull request by fdlai**
## Describe your changes
Previously the 2 buttons below (in the screenshot) were non-functional when editing a blog post.

I've defined a function 'onSubmit' which handles the clicking of those buttons. When calling that function within an edited blog, I'm using useMutation with a graphQL query to update blogs on the server. The other parts of updated code in this branch were necessary to fix several bugs that appeared along the way.

Some of those bugs included: 
1) The blogs not being merged correctly in the cache.
2) The 'keywords' and 'movies' fields, of the blog post, not correctly merging together in the cache.
3) The values from the database for a blog, sometimes not prefilling the form.
4) Not being able to unassociate 'keywords' or 'movies' from a blog.
And several more.

Video showing the functionality:
https://www.loom.com/share/7f64ff52333f4330a4261cffdf617f35

## Issue ticket number and link
slop-56
https://tripleten-apiary.atlassian.net/jira/software/projects/SLOP/boards/2?selectedIssue=SLOP-56

## Checklist before requesting a review

- [x] I have performed a self-review of my code
- [ ] If there are changes to components, I have added / updated automated tests
- [ ] I have made any necessary updates to Storybook to accompany these changes
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings

### CODE REVIEW

File: src/page/blog/review-page.jsx [hunk 1]
```diff
@@ -81,7 +81,10 @@ export const Review = ({ id }) => {
 
   post.keywords.forEach((keyword) => {
     keywordsList.push(
-      <Button className="bg-yellow-button text-sm h-8 w-30 text-black text-center items-center mb-0 py-0 px-2">
+      <Button
+        key={keyword.id}
```
Comment (fdlai) [inline src/page/blog/review-page.jsx:85]: This was a quick unrelated fix along the way. This stops React from complaining about this list lacking a key.

**Review by dani9401 — APPROVED** [review]
Hey Fred! I am marking this as "approved" because I believe the issue I found is related to Slop 31 (see video attached). But let me know if Tomer disagrees for some reason and I an re-review if need be. 

Awesome job on this!

https://github.com/jahorwitz/slopopedia/assets/126746048/42cb34c2-7f06-40ab-b852-3f3466386b4d

**Review by ATribbey — APPROVED** [review]
Works as described!

Well done!

File: src/components/form/form.jsx [hunk 2]
```diff
@@ -260,7 +260,7 @@ Form.Combobox = ({
   setValue,
   ...rest
 }) => {
-  const selectedItems = watch(id) || [];
```
Comment (tomershvueli) [inline src/components/form/form.jsx:263]: Was this change needed?

**Review by tomershvueli — APPROVED** [review]
Working well on my end. Very nicely done!

**Comment by tomershvueli** [comment]
@fdlai let's sync this up with `main` and I can merge it in.

File: src/components/form/form.jsx (hunk 2 above)
Comment (fdlai) [reply src/components/form/form.jsx:263]: I'm can't remember. I can take a look if you want, but I expect @dani9401 's combo box PR to change this code regardless, as it deals with this area directly.

**Comment by fdlai** [comment]
@tomershvueli Ok, I've merged main into this branch!

**Comment by tomershvueli** [comment]
Works great!
